lib/spells/effects/Catapult.h

```diff
@@ -29,6 +30,24 @@ class Catapult : public LocationEffect
 	void serializeJsonEffect(JsonSerializeFormat & handler) override;
 private:
 	int targetsToAttack = 0;
+	//Ballistics percentage
+	int gate = 0;
+	int keep = 0;
+	int tower = 0;
+	int wall = 0;
+	//Damage percentage, used for both ballistics and earthquake
+	int hit = 0;
+	int crit = 0;
+	int noDmg = 0;
+	bool isSeparate = false; //True is like ballistics, false is like earthquake
+	static bool isWallPartAttackable(EWallPart part, const Mechanics * m);
```

Comment:
```suggestion
	bool isWallPartAttackable(EWallPart part, const Mechanics * m) const;
```

Reply:
Why? It should be static, I did this by design. I know your hate against static functions, but this functions is clearly not intended to take this as parameter.

Reply:
Why it should be static? Why it is not intended to take this as parameter?

What if we want to implement "battering ram" spell/ability that can only attacks gates and now have to deal with code that calls this method without instance?

Reply:
@IvanSavenko you do not need to change anything in code to implement battering ram. Just change all probabilities except gates to 0, and gates to 100. This method just clearly should not belongs here, it should belongs to CBattleInfoEssentials, if you want it to take instance. Will move if desired.

Reply:
> This method just clearly should not belongs here, it should belongs to CBattleInfoEssentials

And here is the answer. 
This method does not needs static, instead it should be placed somewhere else.

As I said somewhere - static is often indication of bad design. For example, method that should have been placed somewhere else.

Reply:
moved to CBattleInfoCallback